TombEngine/Objects/TR5/Entity/tr5_roman_statue.cpp

```diff
@@ -74,30 +74,31 @@ namespace TEN::Entities::Creatures::TR5
 		MS_HEAVY_DMG  = 0x10510
 	};
 
-	static void RomanStatueHitEffect(ItemInfo* item, Vector3i* pos, int joint)
+	static void RomanStatueHitEffect(ItemInfo* item, Vector3i& pos, int joint)
 	{
-		*pos = GetJointPosition(item, joint, *pos);
+		pos = GetJointPosition(item, joint, pos);
 
 		if (!(GetRandomControl() & 0x1F))
 		{
-			int fxNumber = CreateNewEffect(item->RoomNumber);
+			int fxNumber = CreateNewEffect(item->RoomNumber, ID_ENERGY_BUBBLES, Pose(pos));
 			if (fxNumber != NO_ITEM)
 			{
-				auto* fx = &EffectList[fxNumber];
-
-				fx->pos.Position = *pos;
-				fx->roomNumber = item->RoomNumber;
-				fx->pos.Orientation.z = 0;
-				fx->pos.Orientation.x = 0;
-				fx->pos.Orientation.y = 2 * GetRandomControl();
-				fx->speed = 1;
-				fx->fallspeed = 0;
-				fx->objectNumber = ID_BODY_PART;
-				fx->color = Vector4::One;
-				fx->flag2 = 9729;
-				fx->frameNumber = Objects[ID_BUBBLES].meshIndex + (GetRandomControl() & 7);
-				fx->counter = 0;
-				fx->flag1 = 0;
+				auto& fx = g_Level.Items[fxNumber];
+				auto& fxInfo = GetFXInfo(fx);
+
+				fx.Pose.Position = pos;
+				fx.RoomNumber = item->RoomNumber;
+				fx.Pose.Orientation.z = 0;
+				fx.Pose.Orientation.x = 0;
+				fx.Pose.Orientation.y = 2 * GetRandomControl();
+				fx.Animation.Velocity.z = 1;
+				fx.Animation.Velocity.y = 0;
+				fx.ObjectNumber = ID_BODY_PART;
+				fx.Model.Color = Vector4::One;
+				fx.Animation.FrameNumber = Objects[ID_BUBBLES].meshIndex + (GetRandomControl() & 7);
```

Comment:
I think what's generally wrong here is that mesh index is stored in Animation.FrameNumber. We now have explicitly stated mesh indices in Model.MeshIndex, this should be used instead.